pkg/services/govmomi/service.go

```diff
@@ -85,29 +86,29 @@ func (vms *VMService) ReconcileVM(ctx *context.VMContext) (vm infrav1.VirtualMac
 		// but sometimes this error is transient, for instance, if the storage was temporarily disconnected but
 		// later recovered, the machine will recover from this error.
 		if wasNotFoundByBIOSUUID(err) {
-			conditions.MarkFalse(ctx.VSphereVM, infrav1.VMProvisionedCondition, infrav1.NotFoundByBIOSUUIDReason, clusterv1.ConditionSeverityWarning, err.Error())
+			conditions.MarkFalse(vmCtx.VSphereVM, infrav1.VMProvisionedCondition, infrav1.NotFoundByBIOSUUIDReason, clusterv1.ConditionSeverityWarning, err.Error())
 			vm.State = infrav1.VirtualMachineStateNotFound
 			return vm, err
 		}
 
 		// Otherwise, this is a new machine and the VM should be created.
 		// NOTE: We are setting this condition only in case it does not exist, so we avoid to get flickering LastConditionTime
 		// in case of cloning errors or powering on errors.
-		if !conditions.Has(ctx.VSphereVM, infrav1.VMProvisionedCondition) {
-			conditions.MarkFalse(ctx.VSphereVM, infrav1.VMProvisionedCondition, infrav1.CloningReason, clusterv1.ConditionSeverityInfo, "")
+		if !conditions.Has(vmCtx.VSphereVM, infrav1.VMProvisionedCondition) {
+			conditions.MarkFalse(vmCtx.VSphereVM, infrav1.VMProvisionedCondition, infrav1.CloningReason, clusterv1.ConditionSeverityInfo, "")
 		}
 
 		// Get the bootstrap data.
-		bootstrapData, format, err := vms.getBootstrapData(ctx)
+		bootstrapData, format, err := vms.getBootstrapData(vmCtx)
 		if err != nil {
-			conditions.MarkFalse(ctx.VSphereVM, infrav1.VMProvisionedCondition, infrav1.CloningFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
+			conditions.MarkFalse(vmCtx.VSphereVM, infrav1.VMProvisionedCondition, infrav1.CloningFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
 			return vm, err
 		}
 
 		// Create the VM.
-		err = createVM(ctx, bootstrapData, format)
+		err = createVM(vmCtx, bootstrapData, format)
```

Comment:
```suggestion
		err = createVM(ctx, vmCtx, bootstrapData, format)
```
Need to refactor function `createVM`